app/ui/browser-modern/views/browser/page/index.jsx

```diff
@@ -73,33 +80,75 @@ class Page extends Component {
       this.props.dispatch(UIEffects.focusURLBar(this.props.pageId, { select: isTofino }));
     });
 
+    this.webview.addEventListener('did-finish-loading', () => {
+      // Event fired only when page is successfully loaded
```

Comment:
If this can be emitted multiple times, then a LOADED state can be set *after* a FAILED state being set.